src/Compilers/CSharp/Test/Semantic/FlowAnalysis/RegionAnalysisTests.cs

```diff
@@ -5843,6 +5987,7 @@ void M(int x)
             Assert.Equal("this, x", GetSymbolNamesJoined(analysis.CapturedOutside));
             Assert.Equal(null, GetSymbolNamesJoined(analysis.DataFlowsIn));
             Assert.Equal(null, GetSymbolNamesJoined(analysis.DataFlowsOut));
+            Assert.Equal("this, x", GetSymbolNamesJoined(analysis.DefinitelyAssignedOnEntry));
```

Comment:
@gafter @agocke @jcouv .  This is the test that doesn't feel correct to me.  For this region, i would expect 'this, x, a, y" to all be definitely assigned here.

Reply:
We don't visit local functions in the usual way, since definite assignment information for them only makes sense on the basis of where they are called from.  We treat the definite assignment state for them as the same as the state on entry to the block in which they are declared, since they could be called earlier.  Not sure what we should do about this.

---
In reply to: [305634988](https://github.com/dotnet/roslyn/pull/37395#discussion_r305634988) [](ancestors = 305634988)

Reply:
This is addressed in [https://github.com/dotnet/roslyn/pull/37450](https://github.com/dotnet/roslyn/pull/37450) which is in review.

---
In reply to: [305926170](https://github.com/dotnet/roslyn/pull/37395#discussion_r305926170) [](ancestors = 305926170,305634988)